ggml.c

```diff
@@ -1122,13 +1122,36 @@ void dequantize_row_q4_1(const void * restrict x, float * restrict y, int k) {
 #define GGML_F16_EPR  8
 
 // F16 arithmetic is not supported by AVX, so we use F32 instead
-// we take advantage of the _mm256_cvt intrinsics to convert F16 <-> F32
 
 #define GGML_F32Cx8             __m256
 #define GGML_F32Cx8_ZERO        _mm256_setzero_ps()
 #define GGML_F32Cx8_SET1(x)     _mm256_set1_ps(x)
+
+#if defined(__F16C__)
+// the  _mm256_cvt intrinsics require F16C
 #define GGML_F32Cx8_LOAD(x)     _mm256_cvtph_ps(_mm_loadu_si128((__m128i *)(x)))
 #define GGML_F32Cx8_STORE(x, y) _mm_storeu_si128((__m128i *)(x), _mm256_cvtps_ph(y, 0))
+#else
+static inline __m256 __avx_f32cx8_load(ggml_fp16_t *x) {
+    float tmp[8];
+
+    for (int i = 0; i < 8; i++)
+        tmp[i] = GGML_FP16_TO_FP32(x[i]);
+
+    return _mm256_loadu_ps(tmp);
+}
+static inline void __avx_f32cx8_store(ggml_fp16_t *x, __m256 y) {
+    float arr[8];
+
+    _mm256_storeu_ps(arr, y);
+
+    for (int i = 0; i < 8; i++)
+        x[i] = GGML_FP16_TO_FP32(arr[i]);
```

Comment:
@anzz1 @ggerganov  Looks like mistake here! Should be `GGML_FP32_TO_FP16()` instead of `GGML_FP16_TO_FP32()`. See that both load and store function right now use same GGML_FP16_TO_FP32(). It is a VERY BAD mistake! And this commit already merged. It affects ONLY cpus that have AVX while not having F16C, hence it is not very noticable mistake.

Reply:
@slaren It is used on my home CPU, on old laptop, that's why I noticed it. My old CPU has AVX, but has no F16C. This combination triggers exactly this path of code. So at least for me your fix would be helpful :)

Reply:
Regardless, should be fixed on master now.

Reply:
@slaren My flags have `-mavx` but DON'T have `-mf16c`. Basically in my CPU I have AVX support, but DON'T have F16C support. That's why this path in code is triggered.

BTW, can you give me URL of pull request that fixes this typo in code, meaning that modifies F16_TO_F32 to be F32_TO_F16? I'd like to pull this fix into my code.